src/Features/CSharp/Portable/Completion/KeywordRecommenders/AndKeywordRecommender.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.KeywordRecommenders
+{
+    internal class AndKeywordRecommender : AbstractSyntacticSingleKeywordRecommender
+    {
+        public AndKeywordRecommender()
+            : base(SyntaxKind.AndKeyword)
+        {
+        }
+
+        protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
+        {
+            return context.IsAfterPatternContext;
```

Comment:
perhaps IsStartPatternContext and IsEndPatternContext.  Or, perhaps: IsAtStartOfPattern and IsAtEndOfPattern.